torch/lib/c10d/FileStore.cpp

```diff
@@ -215,14 +239,16 @@ std::vector<uint8_t> FileStore::get(const std::string& key) {
       std::this_thread::sleep_for(std::chrono::milliseconds(10));
       continue;
     }
-
+    // Always refresh since even though the key exists in the cache,
+    // it might be outdated
     pos_ = refresh(file, pos_, cache_);
+    break;
```

Comment:
Can't break here because it is not guaranteed that the key will be set after this refresh.